ui-tests/cypress/integration/administration/file-browser.spec.js

```diff
@@ -1,92 +1,525 @@
+      currentPage.getContent().getEmptyFolderAlert().should('exist');
+    });
+
+    it([Tag.FEATURE, 'ENG-3297'], 'Navigating out of create text file page NOT to root using breadcrumb', () => {
+      createTestFolder(testFolderInfo);
+      currentPage = openFileBrowserPage();
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().openSubFolder(0);
+      currentPage.getContent().openSubFolder(-1);
+      currentPage.getContent().getFilesTable().should('not.exist');
+      currentPage.getContent().getEmptyFolderAlert().should('exist');
+      currentPage = currentPage.getContent().openCreateTextFilePage();
+      cy.validateAppBuilderUrlPathname('/file-browser/create-text-file');
+      currentPage.getContent().getNameInput().type(textTestFile.name);
+      currentPage.getContent().getTextArea().type(textTestFile.content);
+      currentPage = currentPage.getContent().clickFileBrowserBreadCrumbs(1);
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().getFilesTable().should('exist').and('be.visible');
+      currentPage.getContent().getTableRows().should('have.length', 7);
+      currentPage.getContent().getFolderLink(-1).should('contain', testFolderInfo.name);
+      currentPage.getContent().openSubFolder(-1);
+      currentPage.getContent().getFilesTable().should('not.exist');
+      currentPage.getContent().getEmptyFolderAlert().should('exist');
+    });
+
+    it([Tag.FEATURE, 'ENG-3297'], 'No file is created when canceling out of create text file page', () => {
+      currentPage = openFileBrowserPage();
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().openSubFolder(0);
+      currentPage = currentPage.getContent().openCreateTextFilePage();
+      cy.validateAppBuilderUrlPathname('/file-browser/create-text-file');
+      currentPage.getContent().getNameInput().type(textTestFile.name);
+      currentPage.getContent().getTextArea().type(textTestFile.content);
+      currentPage = currentPage.getContent().clickCancelButton();
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().getFilesTable().should('exist');
+      currentPage.getContent().getTableRows().should('have.length', 6);
+      currentPage.getContent().getFileKebabMenu(`${textTestFile.name}.${textTestFile.extension}`).should('not.exist');
     });
 
   });
 
+  const textTestFile = {name: 'test', extension: 'txt', content: 'this is a test'}
+
+  const testFileInfo          = {path: '',      name: 'data1.json', base64: '', type: 'application/json'}
+  const subfolderTestFileInfo = {path: 'test/', name: 'data1.json', base64: '', type: 'application/json'}
+
+  const createTestFile = (fileInfo) => {
+    cy.fileBrowserController().then(controller => controller.createFile(fileInfo, false))
+      .then(response => {
+        if (response.status == 200) cy.wrap([fileInfo.path + fileInfo.name]).as('filesToBeDeleted')
```

Comment:
if the response is not a successful one (i.e. 2xx) cypress should fail the test immediately; is it not the case?